src/storage/azure.go

```diff
@@ -24,24 +23,25 @@ func NewAzureBlobHandler() (AzureBlobHandler, error) {
 	return AzureBlobHandler{blobClient: blobClient}, nil
 }
 
-func (receiver AzureBlobHandler) FetchFile(sourceUrl string) ([]byte, error) {
+func (receiver AzureBlobHandler) FetchFileByUrl(sourceUrl string) ([]byte, error) {
 	sourceUrlParts, err := azblob.ParseURL(sourceUrl)
 	if err != nil {
 		slog.Error("Unable to parse source URL", slog.String("sourceUrl", sourceUrl), slog.Any(utils.ErrorKey, err))
 		return nil, err
 	}
+	return receiver.FetchFile(sourceUrlParts.ContainerName, sourceUrlParts.BlobName)
+}
 
-	streamResponse, err := receiver.blobClient.DownloadStream(context.Background(), sourceUrlParts.ContainerName, sourceUrlParts.BlobName, nil)
+func (receiver AzureBlobHandler) FetchFile(containerName string, blobName string) ([]byte, error) {
```

Comment:
Do we want to keep `FetchFile` in the `BlobHandler` interface?

Reply:
We refactored the existing fetchFile to become fetchFileByUrl.  So it should be updated in the interface as well. 

We then moved a portion of the old fetch file code into the new fetch file function